packages/angular/cli/src/command-builder/command-runner.ts

```diff
@@ -168,3 +137,23 @@ export async function runCommand(args: string[], logger: logging.Logger): Promis
 
   return process.exitCode ?? 0;
 }
+
+/**
+ * Get the commands that need to be registered.
+ * @returns One or more command factories that needs to be registered.
+ */
+async function getCommandsToRegister(
+  commandName: string | number,
+): Promise<CommandModuleConstructor[]> {
+  const commands: CommandConfig[] = [];
+  if (commandName in RootCommands) {
+    commands.push(RootCommands[commandName as CommandNames]);
+  } else if (commandName in RootCommandsAliases) {
+    commands.push(RootCommandsAliases[commandName]);
+  } else {
+    // Unknown command, register every possible command.
+    Object.values(RootCommands).forEach((c) => commands.push(c));
+  }
+
+  return Promise.all(commands.map((command) => command.factory()));
```

Comment:
nit: Could potentially put the `.then((m) => m.default)` here to avoid repeating it in the factory fields.